src/ts/editorcanvas/CanvasVM.ts

```diff
@@ -133,15 +134,32 @@ module editorcanvas {
             this.systemLoader.loadMap(this.project.projectName, this.data.getEmptyClone())
                 .then((entitySystem : entityframework.EntitySystem) => {
                     this.changeData(entitySystem);
+                    this.loadAssets();
                 }, framework.error.onPromiseError(this._context));
         }
 
+        private loadAssets() {
```

Comment:
Instead of manually checking the number of assets loaded this would be a good place to use Promises and Promise.all